lib/roby/app.rb

```diff
@@ -1041,6 +1047,44 @@ def base_setup
             @base_setup_done = true
         end
 
+        def lock_log_dir
+            return if log_dir_locked?
+
+            return unless @lock_file
+
+            @lock_file ||= File.open(
+                File.join(@log_dir, LOCK_FILE_EXT),
+                File::RDWR | File::CREAT, 0o644
+            )
+            @lock_file.flock(File::LOCK_EX | File::LOCK_NB)
+        rescue Errno::ENOENT, Errno::EACCES, RuntimeError => e
+            warn "Failed to lock directory #{@log_dir}: #{e.message}"
+        end
+
+        def unlock_log_dir
+            return unless log_dir_locked?
+
+            return unless @lock_file
+
+            @lock_file.flock(File::LOCK_UN)
+            @lock_file.close
+            @lock_file = nil
+        rescue Errno::ENOENT, Errno::EACCES, RuntimeError => e
+            warn "Failed to unlock directory #{@log_dir}: " \
+                 "#{e.message}"
+        end
+
+        def log_dir_locked?
```

Comment:
I'm not sure why this method exists.

If `@lock_file` is not nil, the file is locked. Period. Plus, you're opening @lock_file which is itself a File object (not sure what's that supposed to do).